packages/augur-sdk/src/state/Controller.ts

```diff
@@ -31,25 +31,31 @@ export class Controller {
     });
   }
 
-  private updateMarketsData = async (blockNumber: number, allLogs: ParsedLog[]) => {
+  private updateMarketsData = async (marketIds: string[]) => {
+    console.log('marketIds', marketIds.length);
+    const marketsInfo = await Markets.getMarketsInfo(this.augur, await this.db, {
+      marketIds
+    });
+
+    this.augur.events.emit(SubscriptionEventName.MarketsUpdated,  {
+      marketsInfo
+    });
+  };
+
+  private allEvents = async (blockNumber: number, allLogs: ParsedLog[]) => {
     // Grab market ids from all logs.
     // Compose applies operations from bottom to top.
     const logMarketIds = fp.compose(
       fp.compact,
       fp.uniq,
+      fp.filter(NULL_ADDRESS),
       fp.map('market')
     )(allLogs);
 
-    if(logMarketIds.length === 0) return;
-
-    const marketsInfo = await Markets.getMarketsInfo(this.augur, await this.db, {
-      marketIds: logMarketIds
-    });
-
-    this.augur.events.emit(SubscriptionEventName.MarketsUpdated,  {
-      marketsInfo
-    });
-  };
+    if(logMarketIds.length > 0) this.updateMarketsData(logMarketIds as string[]);
+    // emit non market related logs
+    allLogs.forEach(l => l.market === NULL_ADDRESS && this.augur.events.emit(l.name, {...l}));
```

Comment:
If the filter above works we only need to do this:
```suggestion
    allLogs.forEach(l => this.augur.events.emit(l.name, {...l}));
```

Reply:
that change will double send market updated events

Reply:
because market updated events are sent earlier in the code.